sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -121,7 +121,8 @@ class QueryExecutionErrorsSuite extends QueryTest with SharedSparkSession {
       val e1 = intercept[SparkRuntimeException] { lit(v) }
       assert(e1.getErrorClass === "UNSUPPORTED_FEATURE")
       assert(e1.getSqlState === "0A000")
-      assert(e1.getMessage.matches("""The feature is not supported: literal for '.+' of .+\."""))
+      assert(e1.getMessage.matches(
+      """The feature is not supported: a type '.+' is not supported as a literal type."""))
```

Comment:
The goal of SPARK-38097 is to improve error message in the pivot code. This error is fine, and could be the same.